### PR TITLE
[RESTEASY-3018] Vertx add query params to request and improve implementation for Quarkus.

### PR DESCRIPTION
**Pull request by OscarRL**
This implementation makes it easy use the RestEasy Client with Quarkus Vertx.

- Adds a fix to add query params on request.
- Changed uri handling to pass the absolute uri to HttpClient because normal URI fails to resolve port if not specified.
- Check if user agent exists to avoid duplicate agent. 
- Force add vertx handler to use in case that there is no executor service to deserialize request.

### CODE REVIEW

**Review by amollrod — APPROVED** [review]

**Review by amollrod — APPROVED** [review]

**Comment by OscarRL** [comment]
+ Changed uri handling to pass the absolute uri to HttpClient because normal URI fails to resolve port if not specified.

**Review by jamezp — COMMENTED** [review]
We require all contributions to be made under the terms of the ASL 2.0 License: http://www.apache.org/licenses/LICENSE-2.0 
Do you agree to these terms?

**Comment by OscarRL** [comment]
Yes, I expect that. But I did not found any form to apply to the licence on your website.

**Comment by OscarRL** [comment]
Another question is that I have another implementation of the VertX Engine that uses the WebClient extension of VertX 

https://vertx.io/docs/vertx-web-client/java/

This extension is performant as well and requires less code maintenance from Resteasy side and integrates nicely with Quarkus.

We should create another project like resteasy-client-vertx-web-client or we should try to integrate with this one but making webclient dependencies as optional ?

File: resteasy-client-vertx/pom.xml [hunk 1]
```diff
@@ -60,6 +60,11 @@
             <artifactId>junit</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.jboss.resteasy</groupId>
+            <artifactId>resteasy-rxjava2</artifactId>
+            <version>${project.version}</version>
+        </dependency>
```
[the remark this answers is not in the document]
Comment (OscarRL) [reply resteasy-client-vertx/pom.xml]: Solved

**Comment by jamezp** [comment]
> Another question is that I have another implementation of the VertX Engine that uses the WebClient extension of VertX
> 
> https://vertx.io/docs/vertx-web-client/java/
> 
> This extension is performant as well and requires less code maintenance from Resteasy side and integrates nicely with Quarkus.
> 
> We should create another project like resteasy-client-vertx-web-client or we should try to integrate with this one but making webclient dependencies as optional ?

To be completely honest we (I?) have been in purge mode with RESTEasy recently. I think we've got too many projects that are mostly independent of RESTEasy.

Some of these things we've moved to new projects. Some of them we moved to https://github.com/resteasy/resteasy-extensions.

That said I'm definitely not opposed to new things like this. I just think finding the right home is key and I'm absolutely open for discussions about it. We could start a discussion https://github.com/resteasy/resteasy/discussions if you'd like.

**Comment by OscarRL** [comment]
> > Another question is that I have another implementation of the VertX Engine that uses the WebClient extension of VertX
> > https://vertx.io/docs/vertx-web-client/java/
> > This extension is performant as well and requires less code maintenance from Resteasy side and integrates nicely with Quarkus.
> > We should create another project like resteasy-client-vertx-web-client or we should try to integrate with this one but making webclient dependencies as optional ?
> 
> To be completely honest we (I?) have been in purge mode with RESTEasy recently. I think we've got too many projects that are mostly independent of RESTEasy.
> 
> Some of these things we've moved to new projects. Some of them we moved to https://github.com/resteasy/resteasy-extensions.
> 
> That said I'm definitely not opposed to new things like this. I just think finding the right home is key and I'm absolutely open for discussions about it. We could start a discussion https://github.com/resteasy/resteasy/discussions if you'd like.

yes but I think that Red Hat is focusing the development to Quarkus. And a good VertX core for the http should integrate better than the Apache HttpCore 4, that currently is outdated cause Apache released the version 5.

**Comment by jamezp** [comment]
> yes but I think that Red Hat is focusing the development to Quarkus. And a good VertX core for the http should integrate better than the Apache HttpCore 4, that currently is outdated cause Apache released the version 5.

I can definitely say that WildFly is a heavy focus as well :) I have however suggested that we start looking at deprecating the Apache HTTP Client integration in favor of Vert.x. This will like take some time to happen though as we also need to focus on Jakarta REST 3.0 and 3.1.

**Comment by OscarRL** [comment]
Updated master branch and tests solved.

**Comment by jamezp** [comment]
Thank you for the contribution. We are cleaning up older PR's that seem to be stale. If you feel this is still an issue please feel free to re-open.

**Pull request closed without merging**